src/dex/uniswap.js

```diff
@@ -33,23 +45,77 @@ export default class UniswapAdaptor {
     };
     try {
       const blockNumber = await this._provider.getBlockNumber();
-      if (blockNumber === this._lastBlock) return;
-      this._lastBlock = blockNumber;
```

Comment:
I removed this to stop an error where the book was undefined in the caller.  If it's the same block, I see no harm in doing the full `fetch()` again.  It will take a little more time, but the client's view of the seconds passed will have changed, which may result in `take()`.